csharp/private/rules/resgen.bzl

```diff
@@ -0,0 +1,99 @@
+load(
+    "@d2l_rules_csharp//csharp/private:providers.bzl",
+    "CSharpResource",
+)
+
+# Label of the csproj template for ResX compilation
+_TEMPLATE = "@d2l_rules_csharp//csharp/private:rules/ResGen.csproj"
+
+# When compiling the csproj, it will look for the embedded resources related to
+# the path of the csproj on disk. Since the csproj is written by the bazel, it will
+# be in the rule bin directory. If we want to get the relative path to the
+# ResX files, we need to get to a junction/symlink for the source. 
+#
+# The junction/symlink that we are using is 4 directories up (\bazel-out\<sandbox>\bin\<name>\)

```

Comment:
Relying on this seems, likely to break.

Reply:
Agree, but I have had difficulty finding a way of configuring relative paths for the `.resx` files that isn't a bit 'hacky'.

Ideally I'd just go with the copy route, but that doesn't handle embedded files like so:

```xml
  <data name="SetupFiles" type="System.Resources.ResXFileRef, System.Windows.Forms">
    <value>Resources\SetupFiles.zip;System.Byte[], mscorlib, Version=4.0.0.0, ...</value>
  </data>
```